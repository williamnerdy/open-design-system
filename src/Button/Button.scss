@import '../styles/variables';

.button {
  padding: 12px 16px;
  cursor: pointer;
  border: none;
  border-radius: 4px;

  &.disabled,
  &[disabled] {
    cursor: default;
    opacity: 0.4;
  }

  &.primary {
    color: $primary-color-reverse;
    background: $primary-color;
  }

  &.secondary {
    color: $secondary-color-reverse;
    background: $secondary-color;
  }

  &.success {
    color: $success-color-reverse;
    background: $success-color;
  }

  &.danger {
    color: $danger-color-reverse;
    background: $danger-color;
  }

  &.warning {
    color: $warning-color-reverse;
    background: $warning-color;
  }

  &.info {
    color: $info-color-reverse;
    background: $info-color;
  }
}
