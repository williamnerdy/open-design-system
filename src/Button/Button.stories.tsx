import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean, select } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import Button, { Colors } from './Button';

const colors = ['primary', 'secondary', 'success', 'danger', 'warning', 'info'];

storiesOf('Components/Button', module).add('variants', () => (
  <Button
    color={select('Color', colors, 'primary') as Colors}
    onClick={action('clicked')}
    disabled={boolean('Disabled', false)}
  >
    {text('Text', 'Hello Storybook')}
  </Button>
));
