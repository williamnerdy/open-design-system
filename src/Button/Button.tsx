import React from 'react';
import classNames from 'classnames';
import './Button.scss';

export type Colors =
  | 'primary'
  | 'secondary'
  | 'success'
  | 'danger'
  | 'warning'
  | 'info';

interface Props {
  /** Button color */
  color: Colors;
  /** Children */
  children: React.ReactNode;
  /** Button type */
  type?: 'button' | 'submit' | 'reset';
  /** Callback */
  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  /** Whether button is disabled */
  disabled?: boolean;
}

const Button = ({
  color = 'primary',
  type,
  onClick,
  disabled,
  children
}: Props) => (
  <button
    type={type}
    className={classNames('button', color)}
    onClick={onClick}
    disabled={disabled}
  >
    {children}
  </button>
);

export default Button;
