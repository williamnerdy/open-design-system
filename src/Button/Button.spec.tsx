import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Button from './Button';

describe('<Button />', () => {
  afterEach(cleanup);

  it('should render correctly', () => {
    const { asFragment } = render(<Button color="primary">My button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
