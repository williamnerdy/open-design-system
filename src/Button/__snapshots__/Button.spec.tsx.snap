// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
<DocumentFragment>
  <button
    class="button primary"
  >
    My button
  </button>
</DocumentFragment>
`;
